src/components/markdown/ProblemsList/DivisionList/DivisionList.tsx

```diff
@@ -284,6 +285,20 @@ export function DivisionList(props): JSX.Element {
     }
   }, []);
 
+  const problems: DivisionProblemInfo[] =
+    divisionToSeasonToProbs[curDivision][curSeason];
+  const allHavePercent = problems.every(problem => !!problem.percentageSolved);
+
+  const sortOrders = ['By Contest'];
+  if (allHavePercent) sortOrders.push('By Percent');
+  const [sortOrder, setSortOrder] = React.useState('Sort: ' + sortOrders[0]);
+  const getDisplayedProblems = () => {
+    if (!allHavePercent || !sortOrder.endsWith('Percent')) return problems;
+    return [...problems].sort(
+      (a, b) => b.percentageSolved - a.percentageSolved
+    );
+  };
```

Comment:
You can probably improve the performance by wrapping it with [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo). Not sure how necessary it is though